modules/compiler/spirv-ll/test/README.md

```diff
@@ -3,4 +3,44 @@
 To invoke this test suite use the `check-spirv-ll-lit` CMake target to run
 [lit][lit] on the configured test inputs residing in the build directory.
 
+## Modes
+
+You can build this suite in one of two modes: online and offline (the default)
+
+### Offline Mode
+
+In offline mode - the default, or when `CA_ASSEMBLE_SPIRV_LL_LIT_TESTS_OFFLINE`
+is enabled in cmake - the test suite will build its SPIR-V binaries ahead of
+time.
+
+This mode requires `spirv-as` to build the `spvasm` test binaries, and
+`glslangValidator` to build the `glsl` test binaries. Both are optional: if
+either tool is not found the tests will be disabled at runtime.
+
+### Online Mode
+
+In offline mode - when `CA_ASSEMBLE_SPIRV_LL_LIT_TESTS_OFFLINE`
+is disable in cmake - the test suite will build its SPIR-V binaries at test
+runtime. Each test builds its own binary.
+
+This mode is suited to systems where the tools are not necessarily known at
+build time. It also enables quick turn-around time for developers when
+iterating on tests, as there are no dependencies on test execution except the
+test files themselves.
+
+### Switching Between Modes
+
+It is possible to switch between modes at runtime by setting the
+``spirv-ll-online`` parameter:
+
+```
+> ca-lit --param spirv-ll-online=0 modules/compiler/spirv-ll/test
+> ca-lit --param spirv-ll-online=1 modules/compiler/spirv-ll/test
+```
+
+Explicitly enabling the option with a truthy value forcibly enables the online
```

Comment:
Is there any way to make this happen with the check-*-lit target?

Reply:
yes, with `LIT_OPTS` - my mind blanked when writing this and just moved on. Added now.